### PR TITLE
[WIP] Enable CUDA+OMP CVODE strategy

### PR DESCRIPTION
**Pull request by balos1**
... and some other minor fixes I noticed.

To complete this:

- [x] verify solution is reasonable
- [ ] determine why sparse custom linear solver fails with more than 1 thread (@AnneFelden )

### CODE REVIEW

File: Testing/Exec/ReactEval_C/main.cpp [hunk 1]
```diff
@@ -293,7 +293,7 @@ main (int   argc,
     const auto tiling = MFItInfo().SetDynamic(true);
 #pragma omp parallel
 #else
-    const bool tiling = TilingIfNotGPU();
+    const bool tiling = amrex::TilingIfNotGpu();
```
Comment (drummerdoc) [inline Testing/Exec/ReactEval_C/main.cpp]: Not necessary since we are using the AMReX namespace (above)

File: Testing/Exec/ReactEval_C/main.cpp [hunk 2]
```diff
@@ -119,6 +108,10 @@ main (int   argc,
 
     ParmParse ppa("amr");
     ppa.query("plot_file",pltfile);
+    std::vector<double> npts(3);
+    for (int i = 0; i < 3; ++i) {
+        ppa.get("n_cell", npts[i], i);
+    }
```
Comment (drummerdoc) [inline Testing/Exec/ReactEval_C/main.cpp]: Do:
```
Vector<int> npts(AMREX_SPACEDIM);
ppa.getarr("n_cell",npts,0,AMREX_SPACEDIM);
```

File: Reactions/Fuego/GPU/cvode/reactor.cpp [hunk 3]
```diff
@@ -278,6 +275,9 @@ int react(const amrex::Box& box,
             cusolver_status = cusolverSpCreate(&(user_data->cusolverHandle));
             assert(cusolver_status == CUSOLVER_STATUS_SUCCESS);
 
+            cusolver_status = cusolverSpSetStream(user_data->cusolverHandle, stream);
+            assert(cusolver_status == CUSOLVER_STATUS_SUCCESS);
+
```
Comment (drummerdoc) [inline Reactions/Fuego/GPU/cvode/reactor.cpp:276]: Suggest using ``Assert(cusolver_status == CUSOLVER_STATUS_SUCCESS);`` or
AMREX_ALWAYS_ASSERT_WITH_MESSAGE(...,"Here is what happened and how to fix);

File: Reactions/Fuego/GPU/cvode/reactor.cpp [hunk 4]
```diff
@@ -677,6 +677,70 @@ int react(realtype *rY_in,    realtype *rY_src_in,
...
+            BL_PROFILE_VAR("CuSolverInit", CuSolverInit);
+            size_t workspaceInBytes, internalDataInBytes;
+            cusolverStatus_t cusolver_status = CUSOLVER_STATUS_SUCCESS;
+            cusparseStatus_t cusparse_status = CUSPARSE_STATUS_SUCCESS;
+
+            workspaceInBytes = 0;
+            internalDataInBytes = 0;
+
+            cusolver_status = cusolverSpCreate(&(user_data->cusolverHandle));
+            assert(cusolver_status == CUSOLVER_STATUS_SUCCESS);
```
Comment (drummerdoc) [inline Reactions/Fuego/GPU/cvode/reactor.cpp]: Prefer ``Assert`` or ``ALWAYS_ASSERT...``

File: Testing/Exec/ReactEval_C/main.cpp [hunk 5]
```diff
@@ -402,10 +402,13 @@ main (int   argc,
...
-                                   tmp_vect_energy, tmp_src_vect_energy,
-                                   &dt_incr, &time,
-                                   &ode_iE, &ncells, amrex::Gpu::gpuStream());
+                react_1(box,
+                      rhoY, frcExt, T,
+                      rhoE, frcEExt,
+                      fc, mask,
+                      dt_incr, time);
+#endif
+
```
Comment (drummerdoc) [inline Testing/Exec/ReactEval_C/main.cpp]: @EnnaDelfen Is this correct??

File: Testing/Exec/ReactEval_C/inputs.3d-regt [hunk 6]
```diff
@@ -26,11 +26,13 @@ cvode.solve_type = dense
 # Max size of problem
 max_grid_size  = 16
 # Choose name of output pltfile
-amr.plot_file  = plt
+amr.plot_file  = /gpfs/wolf/gen138/scratch/balos1/hackathon/plt
```
Comment (drummerdoc) [inline Testing/Exec/ReactEval_C/inputs.3d-regt]: Why is this here?

File: Testing/Exec/ReactEval_C/main.cpp [hunk 7]
```diff
@@ -91,6 +94,12 @@ main (int   argc,
 
         // domain size
         pp.query("max_grid_size",max_grid_size);
+	
+	// first dim
+        pp.query("first_dim",first_dim);
+        
+	// second dim
+        pp.query("second_dim",second_dim);
```
Comment (drummerdoc) [inline Testing/Exec/ReactEval_C/main.cpp]: ick, yeah lets read in the vector, as one of the later commits does

**Review by drummerdoc — CHANGES_REQUESTED** [review]

File: Testing/Exec/ReactEval_C/inputs.3d-regt (hunk 6 above)
Comment (balos1) [reply Testing/Exec/ReactEval_C/inputs.3d-regt]: Was an old commit, this has since been changed (and the commit squashed).

File: Testing/Exec/ReactEval_C/main.cpp (hunk 5 above)
Comment (balos1) [reply Testing/Exec/ReactEval_C/main.cpp]: @drummerdoc If you look at the current diff in GitHub, these lines haven't changed now. I think it was showing as changed because of a conflicted git merge which I fixed now by rebasing (hence why there are only 2 commits in this PR now).

File: Reactions/Fuego/GPU/cvode/reactor.cpp (hunk 4 above)
Comment (balos1) [reply Reactions/Fuego/GPU/cvode/reactor.cpp]: For what it's worth, `assert` is used everywhere in this file (beyond what I have added). Does someone want to go in and change all of these in one PR?

Comment (drummerdoc) [reply Reactions/Fuego/GPU/cvode/reactor.cpp]: Yeah, leave it.  I'll fix later.
